### PR TITLE
Fix kube-proxy panic when running with "--cleanup-iptables=true"

### LINKED ISSUE #48177: hyperkube kube-proxy panics when run with '--cleanup-iptables=true' starting with v1.7.0-alpha.3
**Is this a BUG REPORT or FEATURE REQUEST?**:
bug

**What happened**:
Got panic after running:
```
docker run --net=host --rm=true --privileged --volume=/var/run/dbus gcr.io/google_containers/hyperkube-amd64:v1.7.0-rc.1 /hyperkube proxy --cleanup-iptables=true
```

Panic:
```
2017-06-28 06:19:08.012844 I | proto: duplicate proto type registered: google.protobuf.Any
2017-06-28 06:19:08.044749 I | proto: duplicate proto type registered: google.protobuf.Duration
2017-06-28 06:19:08.045078 I | proto: duplicate proto type registered: google.protobuf.Timestamp
W0628 06:19:08.165958       1 server.go:188] WARNING: all flags other than --config, --write-config-to, and --cleanup-iptables are deprecated. Please begin using a config file ASAP.
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x3747724]

goroutine 1 [running]:
k8s.io/kubernetes/cmd/kube-proxy/app.(*ProxyServer).Run(0xc42069c240, 0xc420884d01, 0xc4203e7700)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/cmd/kube-proxy/app/server.go:630 +0x74
k8s.io/kubernetes/cmd/kube-proxy/app.(*Options).Run(0xc420884d10, 0x0, 0x0)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/cmd/kube-proxy/app/server.go:226 +0xa0
k8s.io/kubernetes/cmd/kube-proxy/app.NewProxyCommand.func1(0xc4205efd40, 0x92146d8, 0x0, 0x0)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/cmd/kube-proxy/app/server.go:341 +0x80
k8s.io/kubernetes/vendor/github.com/spf13/cobra.(*Command).execute(0xc4205efd40, 0xc420281220, 0x0, 0x1, 0xc4205efd40, 0xc420281220)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/github.com/spf13/cobra/command.go:603 +0x22b
k8s.io/kubernetes/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc4205efd40, 0xc420a579e8, 0x42be9e, 0xc4200ecca8)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/github.com/spf13/cobra/command.go:689 +0x339
k8s.io/kubernetes/vendor/github.com/spf13/cobra.(*Command).Execute(0xc4205efd40, 0xc420281220, 0xc420a579f8)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/github.com/spf13/cobra/command.go:648 +0x2b
main.NewKubeProxy.func1(0xc4200787e0, 0xc420281220, 0x0, 0x1, 0x0, 0x0)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/cmd/hyperkube/kube-proxy.go:50 +0x68
main.(*HyperKube).Run(0xc420084a80, 0xc420010190, 0x1, 0x1, 0x0, 0x0)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/cmd/hyperkube/hyperkube.go:177 +0x3ec
main.(*HyperKube).RunToExit(0xc420084a80, 0xc420010180, 0x3, 0x3)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/cmd/hyperkube/hyperkube.go:187 +0x59
main.main()
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/cmd/hyperkube/main.go:47 +0x847
```

**What you expected to happen**:
No panic and exit successfully.

**How to reproduce it (as minimally and precisely as possible)**:

Run the following command on both CoreOS v1298.7.0(bare metal) and Ubuntu 16.04.2 LTS(OpenStack)
```
docker run --net=host --rm=true --privileged --volume=/var/run/dbus gcr.io/google_containers/hyperkube-amd64:v1.7.0-rc.1 /hyperkube proxy --cleanup-iptables=true
```
Happened with the following versions:
v1.7.0-alpha.3
v1.7.0-alpha.4
v1.7.0-beta.0
v1.7.0-beta.1
v1.7.0-beta.2
v1.7.0-rc.1
v1.8.0-alpha.0
v1.8.0-alpha.1

**Anything else we need to know?**:
I guess it caused by 43cb024402e6992418b2197228889633b82ddb75.

**Environment**:
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration**:
- OS (e.g. from /etc/os-release):
Both CoreOS v1298.7.0(bare metal) and Ubuntu 16.04.2 LTS(OpenStack)
- Kernel (e.g. `uname -a`):
  - CoreOS v1298.7.0: 
     Linux node-f7d74e 4.9.16-coreos-r1 #1 SMP Tue Apr 4 19:09:38 UTC 2017 x86_64 Intel(R) Core(TM) i5-6500 CPU @ 3.20GHz GenuineIntel GNU/Linux
  - Ubuntu 16.04.2 LTS: 
     Linux ubuntu-3b1877 4.4.0-62-generic #83-Ubuntu SMP Wed Jan 18 14:10:15 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by xiangpengzhao**
**What this PR does / why we need it**:
Save the `--cleanup-iptables` config for new ProxyServer when it's true.
Also check if Broadcaster and EventClient are nil to avoid other potential panic.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #48177

**Special notes for your reviewer**:
/cc @ncdc @irake99

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: cmd/kube-proxy/app/server.go [hunk 1]
```diff
@@ -447,7 +447,7 @@ func NewProxyServer(config *componentconfig.KubeProxyConfiguration, cleanupAndEx
 
 	// We omit creation of pretty much everything if we run in cleanup mode
 	if cleanupAndExit {
-		return &ProxyServer{IptInterface: iptInterface}, nil
+		return &ProxyServer{IptInterface: iptInterface, CleanupAndExit: cleanupAndExit}, nil
```
Comment (xiangpengzhao) [inline cmd/kube-proxy/app/server.go]: To reviewer: this will guarantee the server to exit when `--cleanup-iptables=true`, see code block L603-L610.

**Comment by luxas** [comment]
cc @kubernetes/kubernetes-release-managers

**Comment by caesarxuchao** [comment]
I assume it should be cherrypicked.

**Comment by ncdc** [comment]
> I assume it should be cherrypicked.

Yes

**Comment by xiangpengzhao** [comment]
>I assume it should be cherrypicked.

Seems like this bug was introduced in #34727 whose milestone is also v1.7. So I think we don't need to cherry-pick it to a former release. @ncdc could you help confirm this? Thanks!

**Comment by ncdc** [comment]
Need to cherry pick to release-1.7 branch

**Comment by xiangpengzhao** [comment]
Hmm, I thought this will be merged to v1.7 directly:)

It's deep night at my local time. I will go back to office about 5 hours later and have the chance to do the cheey-picking. If this is urgent, can @ncdc or other folks help do it? Thanks!

**Comment by caesarxuchao** [comment]
@xiangpengzhao I'll do a batch cherrypick. Thanks for the fix.

**Comment by xiangpengzhao** [comment]
Thanks @caesarxuchao!

**Pull request merged**